### DISCUSSION TITLE
Help with usage scenario

### DISCUSSION THREAD

**Discussion by MarkoSchuetz** [Xpra-org/xpra-html5#220] [General]
A little background: on a Linux laptop I'm using noVNC and x0vncserver to remotely access my desktop from an Android tablet. The Android tablet has Nreal Air glasses attached to it and in Nebula (the AR software that comes with the glasses) I can then connect from the browser to the desktop and have the (huge) desktop fixated in AR space.
So far, so good. But the lag for displaying any updates (specifically mouse movements) is quite annoying. I was hoping to set something up with Xpra and that that would have less lag, but have so far run into trouble. When the entire desktop gets exported using shadow, then the in-browser display on the Android side is not updated e.g. when popping up yakuake. I also tried starting a separate Xpra process for a PDF viewer, but then I need to control it from the keyboard and mouse attached to the laptop instead of from any virtual keyboard on the tablet and (at least as far as I have found out until now) I cannot do that, or would there to a way to do so?

I was hoping that someone here would have some great advice...

Thanks in advance,

Marko

**Comment by totaam** [comment]
@MarkoSchuetz - not sure what version you have installed, but screen updates should always be propagated.
`shadow` mode in inherently slow, have you considered using a dedicated desktop or seamless session to run this app?

**Reply by MarkoSchuetz** [reply]
If I understand you correctly then by "seamless" you mean just using`xpra start`. I did that but remember that I need to control the running application using the keyboard and mouse attached to the laptop (the "remote").
Having a dedicated desktop seems to imply that I could not just attach to the open session that I have on the laptop. Also, I suspect that some open applications would not deal well with a second instance being launched.

Anyway, I feel I'm still at the beginning of the exploration...

**Reply by MarkoSchuetz** [reply]
Oh, with a dedicated desktop I still would have the issue that I want to use the keyboard and mouse connected to the remote...

**Reply by totaam** [reply]
Can't you attach to the app running as a seamless app from both the laptop and your client? (see `sharing` option).
`seamless` and `desktop` modes are a lot more efficient than `shadow`.

**Reply by MarkoSchuetz** [reply]
I had not thought of this. It gets me closer to what I want... but I see two new issues:

1. running `xpra start --sharing=yes  --html=on --bind-tcp=192.168.xxx.xxx:14501 --start-via-proxy=no --start='zathura Test.pdf'` and then attaching from the laptop and the tablet I don't see the mouse movements on the tablet
2. when paging through the PDF with the keyboard of the laptop the display on the tablet gets updated for several pages, but then the updates stop and in the server log I find:
```
2023-02-16 09:33:01,801 Warning: client decoding error:
2023-02-16 09:33:01,801  DataCloneError: Failed to execute 'postMessage' on 'Worker': Value at index 0 does not have a transfe
rable type.
2023-02-16 09:33:26,127 Warning: delayed region timeout
2023-02-16 09:33:26,127  region is 15 seconds old, will retry - bad connection?
2023-02-16 09:33:26,127  20 late responses:
2023-02-16 09:33:26,127      19 webp :  24s
2023-02-16 09:33:26,127      20 webp :  24s
2023-02-16 09:33:26,127      21 webp :  24s
2023-02-16 09:33:26,127      22 webp :  24s
2023-02-16 09:33:26,128      23 webp :  24s
2023-02-16 09:33:26,128      24 webp :  24s
2023-02-16 09:33:26,128      25 webp :  24s
2023-02-16 09:33:26,128      26 webp :  24s
2023-02-16 09:33:26,128      27 jpeg :  24s
...
```

**Comment by totaam** [comment]
> and then attaching from the laptop and the tablet I don't see the mouse movements on the tablet

When sharing, mouse movements are not mirrored to the other connected clients - though this would be easy enough to add, feel free to create a feature request.

> but then the updates stop and in the server log I find

That's fixed in 5608f6910a609c9c17c4939ec90d22ae8e049a6c
Either apply this fix yourself, downgrade to an older version or wait for the next bug fix release.

**Reply by MarkoSchuetz** [reply]
I created a feature request.

**Reply by totaam** [reply]
#229
